datumaro/components/visualizer.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import warnings
+from typing import Iterable, List, Optional, Tuple, Union
+
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from datumaro.components.annotation import Annotation, AnnotationType, Bbox, LabelCategories
+from datumaro.components.dataset import IDataset
+from datumaro.components.extractor import DatasetItem
+
+DEFAULT_COLOR_CYCLES: List[str] = [
+    "#1f77b4",
+    "#ff7f0e",
+    "#2ca02c",
+    "#d62728",
+    "#9467bd",
+    "#8c564b",
+    "#e377c2",
+    "#7f7f7f",
+    "#bcbd22",
+    "#17becf",
+]
+
+TEXT_Y_OFFSET = 1.0
+
+
+class Visualizer:
+    def __init__(
+        self,
+        dataset: IDataset,
+        ignored_types: Optional[Iterable[AnnotationType]] = None,
+        figsize: Tuple[float, float] = (8, 6),
+        color_cycles: Optional[List[str]] = None,
+        bbox_linewidth: float = 1.0,
+        text_y_offset: float = 1.5,
+    ) -> None:
+        self.dataset = dataset
+        self.figsize = figsize
+        self.ignored_types = set(ignored_types) if ignored_types is not None else set()
+        self.color_cycles = color_cycles if color_cycles is not None else DEFAULT_COLOR_CYCLES
+        self.bbox_linewidth = bbox_linewidth
+        self.text_y_offset = text_y_offset
+
+    def _sort_by_z_order(self, annotations: List[Annotation]) -> List[Annotation]:
+        def _sort_key(ann: Annotation):
+            z_order = getattr(ann, "z_order", -1)
+            return z_order
+
+        return sorted(annotations, key=_sort_key)
+
+    def vis_gallery(
+        self,
+        ids: List[Union[str, DatasetItem]],
+        nrows: int,
+        ncols: int,
```

Comment:
How about setting (nrows, ncols) as a tuple?
Couldn't be optional variable? From the given length of ids, just set ncols to be a given number when it is not specified. 
For instance, when ncols and nrows are not given and the len(ids)=20, we can easily draw the gallery by (5,4) where a predefined ncols=4. What do you think?

Reply:
Added auto-infer grid shape mode for `None` argument.